raiden-ts/tests/unit/epics/monitor.spec.ts

```diff
@@ -357,6 +369,268 @@ describe('monitorRequestEpic', () => {
+        rateToSvt: { [token]: WeiPerEther as UInt<32> },
+      }),
+    );
+
+    userDepositContract.functions.effectiveBalance.mockResolvedValue(monitoringReward.sub(1));
+    await waitBlock();
+
+    expect(raiden.output).toContainEqual(
+      udcDeposit.success(undefined, { totalDeposit: monitoringReward.sub(1) as UInt<32> }),
+    );
+
+    await ensureChannelIsDeposited([raiden, partner], deposit);
+    await ensureChannelIsDeposited([partner, raiden], deposit);
+    await ensureTransferUnlocked([partner, raiden], amount);
+    await waitBlock();
+
+    expect(raiden.output).not.toContainEqual(
+      messageGlobalSend(
+        { message: expect.objectContaining({ type: MessageType.MONITOR_REQUEST }) },
+        expect.anything(),
+      ),
+    );
+  });
+
+  test('ignore: config.monitoringReward unset', async () => {
+    expect.assertions(1);
+    const [raiden, partner] = await makeRaidens(2);
+    raiden.store.dispatch(raidenConfigUpdate({ monitoringReward: null }));
+
+    await ensureChannelIsDeposited([raiden, partner], deposit);
+    await ensureChannelIsDeposited([partner, raiden], deposit);
+    await ensureTransferUnlocked([partner, raiden], amount);
+    await waitBlock();
+
+    expect(raiden.output).not.toContainEqual(
+      messageGlobalSend(
+        { message: expect.objectContaining({ type: MessageType.MONITOR_REQUEST }) },
+        expect.anything(),
+      ),
+    );
+  });
+
+  test('ignore: signing rejected not fatal', async () => {
+    expect.assertions(1);
+
+    const [raiden, partner] = await makeRaidens(2);
+    const monitoringReward = bigNumberify(5) as UInt<32>;
+    raiden.store.dispatch(
+      raidenConfigUpdate({
+        httpTimeout: 30,
+        monitoringReward,
+        caps: {
+          [Capabilities.NO_DELIVERY]: true,
+          [Capabilities.NO_MEDIATE]: true,
+          // 'noReceive' should be auto-disabled by 'getCaps$'
+        },
+        // rate=WeiPerEther == 1:1 to SVT
+        rateToSvt: { [token]: WeiPerEther as UInt<32> },
+      }),
+    );
+    await ensureChannelIsDeposited([raiden, partner], deposit);
+    await ensureChannelIsDeposited([partner, raiden], deposit);
+    await ensureTransferPending([partner, raiden], amount);
+    // fails only after transfer's signatures
+    const originalSign = raiden.deps.signer.signMessage;
+    const signerSpy = jest
+      .spyOn(raiden.deps.signer, 'signMessage')
+      .mockImplementation(async (message) => {
+        if (signerSpy.mock.calls.length > 1) throw new Error('Signature rejected');
+        return originalSign.call(raiden.deps.signer, message);
+      });
+
+    await ensureTransferUnlocked([partner, raiden], amount);
+    await waitBlock();
+
+    expect(raiden.output).not.toContainEqual(
+      messageGlobalSend(
+        { message: expect.objectContaining({ type: MessageType.MONITOR_REQUEST }) },
+        expect.anything(),
+      ),
+    );
+    // expect(signerSpy).toHaveBeenCalledTimes(2);
```

Comment:
And also ensure that `signerSpy` got called enough to reach the throw part, and didn't shutdown, or else it could shutdown because of an unhandled error and the test wouldn't notice.

Reply:
Took care in next fixup commit